src/lerna_manifest.rs

```diff
@@ -90,11 +90,11 @@ impl LernaManifest {
     }
 
     pub fn into_package_manifests_by_package_name(
-        mut self,
+        self,
     ) -> Result<HashMap<String, PackageManifest>, Box<dyn Error>> {
         self.internal_package_manifests
-            .drain(0..)
-            .map(|manifest| Ok((manifest.contents.name.to_owned(), manifest)))
+            .into_iter()
```

Comment:
I hated the `drain` when I wrote it so I'm happy to see this improvement! But if we're using `into_iter` to yield `PackageManifest`, how does this function not have to take  `mut self`?